examples/guards/lib/main.dart

```diff
@@ -224,11 +225,11 @@ class MyApp extends StatelessWidget {
       ],
     ),
     guards: [
-      // Redirect to /login if the user is not authenticated:
+      // Guard /books/* by beaming to /login if the user is unauthenticated:
       BeamGuard(
-        pathBlueprints: ['/books*'],
+        pathBlueprints: ['/books/*'],
         check: (context, location) =>
-            Provider.of<AuthenticationNotifier>(context).isAuthenticated,
+            context.watch<AuthenticationNotifier>().isAuthenticated,
```

Comment:
I believe this should be `context.read`

Reply:
Why not `.watch`? I made it that way because before with `Provider.of<AuthenticationNotifier>(context)` it also listened and it worked.

Reply:
I know, it works.

But for just reading the value in some callback, `read` is the prefered method as it will not bind itself to the given `context`, which we don't need because this is a one-time call.

`watch` is when we want the corresponding context scope to react to changes in Provider, when we build `Widget`s and such.

Reply:
You are right 👍 I will change it.